project/src/main/world/creature-spawner.gd

```diff
@@ -0,0 +1,82 @@
+extends Node2D
+"""
+Conditionally spawns a creature on the overworld.
+
+The decision to spawn the creature is controlled by the 'spawn_if' property.
+
+The creature's creature_id, properties and groups (the 'chattables' group in particular) can be managed by the
+target_properties and target_groups fields.
+"""
+
+export (NodePath) var overworld_world_path: NodePath
+
+# (optional) path to a stool the spawned creature sits on
+export (NodePath) var stool_path: NodePath
+
+# the properties of the spawned creature
+export (Dictionary) var target_properties: Dictionary
+
+# the groups of the spawned creature
+export (Array) var target_groups: Array
+
+# a boolean expression which, if evaluated to 'true', will result in the creature being spawned
+export (String) var spawn_if: String
+
+export (float) var max_fatness := 10.0
+
+var CreaturePackedScene: PackedScene = load("res://src/main/world/creature/Creature.tscn")
+
+# stool the spawned creature sits on, if any
+var _stool: Stool
+
+onready var _overworld_world: OverworldWorld = get_node(overworld_world_path)
+
+func _ready() -> void:
+	if stool_path:
+		_stool = get_node(stool_path)
```

Comment:
Does get_node(null) return null, with no error? If so, `_stool` can be an onready variable.

Reply:
get_node(null) returns an error -- this logic is necessary. There is a `get_node_or_null` but that will conceal misconfigured/nonexistent node paths.